src/test/java/io/confluent/connect/jdbc/source/integration/JdbcSourceConnectorIT.java

```diff
@@ -0,0 +1,249 @@
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.DockerComposeContainer;
+
+import java.io.File;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+@Category(IntegrationTest.class)
+public class JdbcSourceConnectorIT extends BaseConnectorIT {
+  private static final Logger log = LoggerFactory.getLogger(JdbcSourceConnectorIT.class);
+
+  private Map<String, String> props;
+  private static Connection connection;
+
+  @ClassRule
+  public static DockerComposeContainer compose =
+          new DockerComposeContainer(new File("src/test/docker/docker-compose.yml"));
+
+  @Before
+  public void setup() throws Exception {
+    startConnect();
+    connect.kafka().createTopic(KAFKA_TOPIC, 1);
+    props = getCommonProps();
+    if (connection == null) {
+      TestUtils.waitForCondition(() -> assertConnection().orElse(false),
+              TimeUnit.SECONDS.toMillis(30),
+              "Failed to start the container.");
+    }
+  }
+
+  @After
+  public void close() {
+    // delete connector
+    connect.deleteConnector(CONNECTOR_NAME);
+    connect.stop();
+  }
+
+  @AfterClass
+  public static void closeConnection() {
+    compose.close();
+  }
+
+  private Optional<Boolean> assertConnection() {
+    try {
+      connection = getConnection();
+      if (connection != null) {
+        return Optional.of(true);
+      }
+      //Delay to avoid frequent connection requests.
+      TimeUnit.MILLISECONDS.sleep(100);
+      return Optional.empty();
+    } catch (SQLException | InterruptedException e) {
+      return Optional.empty();
+    }
+  }
+
+  @Test
+  public void testTaskBounce() throws Exception {
+    int startIndex = 0;
+
+    dropTableIfExists(KAFKA_TOPIC, connection);
+    createTable(connection);
+    sendTestDataToMysql(startIndex, connection);
+
+    connect.kafka().createTopic("sql-" + KAFKA_TOPIC);
+
+    // Add mysql dialect related configurations.
+    addConnectionProperties(props);
+
+    props.put("mode", "incrementing");
+    props.put("incrementing.column.name",  "id");
+
+    // start a source connector
+    connect.configureConnector(CONNECTOR_NAME, props);
+
+    // wait for tasks to spin up
+    int expectedNumTasks = Integer.valueOf(MAX_TASKS); // or set to actual number
+    waitForConnectorToStart(CONNECTOR_NAME, expectedNumTasks);
+
+    log.info("Waiting for records in destination topic ...");
+    ConsumerRecords<byte[], byte[]> records = connect.kafka().consume(
+        NUM_RECORDS,
+        CONSUME_MAX_DURATION_MS,
+        "sql-mysqlTable");
+
+    assertRecordOffset(startIndex, startIndex+NUM_RECORDS, records);
+
+    // restart Connector Task
+    if(restartConnectorTask() != HttpStatus.SC_NO_CONTENT){
+      throw new Exception("Unable to restart the task");
```

Comment:
handle the exception in the restart method and put a try only on execute method and catch IOException and throw the a RuntimeException with the same message `Unable to restart Connector task.`

Reply:
Try run the connector for greater number of records so that task restarts when connector is processing records.